### PR TITLE
[release/6.0][wasm] Change dotnet.wasm link optimization

### PR DESCRIPTION
**Pull request by radekdoulik**
Context: https://github.com/dotnet/runtime/issues/60349

When we switched to Emscripten 2.0.21, we stopped using deprecated
`--llvm-opts 2` option. This caused unwanted performance degradation.
Switching to `-O2` get us similar performance as with `--llvm-opts`.

Blazor wasm test app `Time to first UI` benchmark times:

    branch/commit        link option    time    dotnet.wasm size  .br size
    --------------------+------------+--------+------------------+---------
    release/6.0 92ff02       -Oz       491ms    2,430,639          850,109
    release/6.0 92ff02       -O2       457ms    2,474,518          853,765
    release/6.0 92ff02       -O3       444ms    2,555,824          857,961

## Customer Impact

Fixes a startup time regression for blazorwasm between .NET 5 an .NET 6 by a reverting compiler option change that was accidentally introduced as part of other build changes.

## Testing

Manual and Automated

## Regression

Yes, startup time regression.

## Risk

Low, switch to compiler options we're using in main.

### CODE REVIEW

**Comment by lewing** [comment]
we're going to let this sit for a bit before we take it in servicing

File: src/mono/wasm/Makefile
Comment (radical) [inline src/mono/wasm/Makefile:75]: Is this needed on `main` too?

Comment (lewing) [reply src/mono/wasm/Makefile:75]: It could go there but it isn't used as part of the normal build

Comment (radekdoulik) [reply src/mono/wasm/Makefile:75]: We don't use these flags in main anymore. I should open PR to remove this variable.

**Comment by Anipik** [comment]
@marek-safar @radical can you review this one ?

**Review by radical — APPROVED** [review]
LGTM. but I would also wait for an approval from @lewing

**Review by marek-safar — APPROVED** [review]

**Review by lewing — APPROVED** [review]

**Pull request merged**
